pkg/segment/writer/segwriter.go

```diff
@@ -1042,29 +1044,76 @@ func BackFillPQSSegmetaEntry(segsetkey string, newpqid string) {
 			continue
 		}
 	}
-	wfd, err := os.OpenFile(localSegmetaFname, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0644)
+	err = WriteSegMeta(localSegmetaFname, preservedSmEntries)
 	if err != nil {
-		log.Errorf("BackFillPQSSegmetaEntry: Failed to open SegMetaFile name=%v, err:%v", localSegmetaFname, err)
-		return
+		log.Errorf("BackFillPQSSegmetaEntry: failed to write Segmeta: err=%v", err)
+	}
+}
```

Comment:
nit: I'd recommend actually keeping the `return` in this error block, even though it is the last part of this function so it doesn't make any difference currently. My reasoning is that later if we add more logic to this function, we will want the `return` here if we get an error

Reply:
addressed in the latest commit.